### PR TITLE
[oneDNN-aarch64] Update threadpool scheduler to make local thread act…

### PR DESCRIPTION
**Pull request by cfRod**
…ivate and deactivate tp

Resolves segmentation fault in https://github.com/tensorflow/tensorflow/pull/58071/

Signed-off-by: cfRod <crefeda.rodrigues@arm.com>

### CODE REVIEW

**Comment by cfRod** [comment]
@penpornk

File: tensorflow/workspace2.bzl [hunk 1]
```diff
@@ -201,7 +201,7 @@ def _tf_repositories():
     tf_http_archive(
         name = "mkl_dnn_acl_compatible",
         build_file = "//third_party/mkl_dnn:mkldnn_acl.BUILD",
-        patch_file = ["//third_party/mkl_dnn:onednn_acl_threadcap.patch", "//third_party/mkl_dnn:onednn_acl_fixed_format_kernels.patch", "//third_party/mkl_dnn:onednn_acl_depthwise_convolution.patch"],
+	patch_file = ["//third_party/mkl_dnn:onednn_acl_threadcap.patch", "//third_party/mkl_dnn:onednn_acl_fixed_format_kernels.patch", "//third_party/mkl_dnn:onednn_acl_depthwise_convolution.patch", "//third_party/mkl_dnn:onednn_reorder_padded.patch", "//third_party/mkl_dnn:onednn_reorder_update.patch","//third_party/mkl_dnn:onednn_acl_threadpoolscheduler.patch"],
```
Comment (penpornk) [inline tensorflow/workspace2.bzl:204]: Nit: The list is getting long, let's make it multi-line.
```suggestion
	patch_file = [
		"//third_party/mkl_dnn:onednn_acl_threadcap.patch", 
		"//third_party/mkl_dnn:onednn_acl_fixed_format_kernels.patch", 
		"//third_party/mkl_dnn:onednn_acl_depthwise_convolution.patch", 
		"//third_party/mkl_dnn:onednn_reorder_padded.patch", 
	    "//third_party/mkl_dnn:onednn_reorder_update.patch","//third_party/mkl_dnn:onednn_acl_threadpoolscheduler.patch"
	],
```

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you very much for the fix, @cfRod!

**Comment by cfRod** [comment]
I noticed that the Arm CI failed because two patches "//third_party/mkl_dnn:onednn_reorder_padded.patch",  and "//third_party/mkl_dnn:onednn_reorder_update.patch", are not on upstream TensorFlow. I will updated the patch

File: tensorflow/workspace2.bzl [hunk 2]
```diff
@@ -201,8 +201,13 @@ def _tf_repositories():
...
         name = "mkl_dnn_acl_compatible",
         build_file = "//third_party/mkl_dnn:mkldnn_acl.BUILD",
-        patch_file = ["//third_party/mkl_dnn:onednn_acl_threadcap.patch", "//third_party/mkl_dnn:onednn_acl_fixed_format_kernels.patch", "//third_party/mkl_dnn:onednn_acl_depthwise_convolution.patch"],
-        sha256 = "fc2b617ec8dbe907bb10853ea47c46f7acd8817bc4012748623d911aca43afbb",
+        patch_file = [
+	"//third_party/mkl_dnn:onednn_acl_threadcap.patch", 
+	"//third_party/mkl_dnn:onednn_acl_fixed_format_kernels.patch", 
+	"//third_party/mkl_dnn:onednn_acl_depthwise_convolution.patch", 
+	"//third_party/mkl_dnn:onednn_acl_threadpoolscheduler.patch"
+		],
```
Comment (penpornk) [inline tensorflow/workspace2.bzl:209]: I think the formatting here looks weird, but let's see if the formatter presubmit will complain.

**Review by penpornk — APPROVED** [review]
Thank you again for the fix!

**Comment by penpornk** [comment]
@cfRod @elfringham Do you know why the ARM CI is skipped here? (This was after I had explicitly clicked "Rerun all jobs" on the CI page.) Should I create a duplicate PR just to test? Thank you!

**Comment by elfringham** [comment]
@penpornk The ARM_CI job was actually scheduled twice, once for each label added 'kokoro:force-run' and 'ready to pull'. The one that shows here is for 'ready to pull' which it is set to ignore. The ARM_CI run for 'kokoro:force-run' is, as I write this, still running. https://github.com/tensorflow/tensorflow/actions/runs/3937186516.
It is unfortunate that you cannot see both ARM_CI jobs in this PR's status box.

**Comment by penpornk** [comment]
@elfringham Thank you very much for the explanation!

File: tensorflow/workspace2.bzl (hunk 2 above)
Comment (cfRod) [reply tensorflow/workspace2.bzl:209]: Okay!

Comment (cfRod) [reply tensorflow/workspace2.bzl:209]: Is there anything that needs to be changed?

Comment (penpornk) [reply tensorflow/workspace2.bzl:209]: No. It got a conflict so I manually fixed it internally today. Just waiting for morning PT so other folks could help approve my changes. (I renamed the patch to `onednn_acl_threadpool_scheduler.patch`, btw.) Thank you for checking!

**Pull request merged**